test/testHelpers.hpp

```diff
@@ -603,46 +623,74 @@ std::string printContext(const std::vector<T>& hGold, std::string goldName,
         os << ", " << coords[3];
     os << "), dims are " << minimalDim4(arrDims, arrDims) << "\n";
 
+    std::vector<int> valsWidths;
+    std::vector<std::string> ctxDim0;
+    std::vector<std::string> ctxOutVals;
+    std::vector<std::string> ctxGoldVals;
+
+    // Get dim0 positions and out/reference values for the context window
+    // Also get the max string length between the position and out/ref values
+    //  per item so that it can be used later as the field width for
+    //  displaying each item in the context window
+    for (uint i = 0; i < (dim0End - dim0Start); ++i) {
+        std::ostringstream tmpOs;
+
+        uint dim0 = dim0Start + i;
+        if (dim0 == coords[0])
+            tmpOs << "[" << dim0 << "]";
+        else
+            tmpOs << dim0;
+        ctxDim0.push_back(tmpOs.str());
+        int dim0Len = tmpOs.str().length();
+        tmpOs.str(std::string());
+
+        uint valIdx = vecStartIdx + i;
+
+        T outVal = hOut[valIdx];
+        if (valIdx == idx)
+            tmpOs << "[" << +hOut[valIdx] << "]";
+        else
+            tmpOs << +hOut[valIdx];
+        ctxOutVals.push_back(tmpOs.str());
+        int outLen = tmpOs.str().length();
+        tmpOs.str(std::string());
+
+        T goldVal = hGold[valIdx];
+        if (valIdx == idx)
+            tmpOs << "[" << +hGold[valIdx] << "]";
+        else
+            tmpOs << +hGold[valIdx];
+        ctxGoldVals.push_back(tmpOs.str());
+        int goldLen = tmpOs.str().length();
+        tmpOs.str(std::string());
+
+        int maxWidth = std::max<int>(dim0Len, outLen);
+        maxWidth = std::max<int>(maxWidth, goldLen);
+        valsWidths.push_back(maxWidth);
+    }
+
     int varNameWidth = std::max<int>(goldName.length(), outName.length());
-    int valsWidth = 10;
 
-    // Display dim0 positions
+    // Display dim0 positions, output values, and reference values
+
     os << std::setw(varNameWidth) << "" << "    ";
-    for (uint i = dim0Start; i < dim0End; ++i) {
-        if (i == coords[0]) {
-            std::ostringstream tmpOs;
-            tmpOs << "[" << i << "]";
-            os << std::setw(valsWidth) << std::left << tmpOs.str() << " ";
-        } else
-            os << std::setw(valsWidth) << std::left << i << " ";
+    for (uint i = 0; i < (dim0End - dim0Start); ++i) {
+        os << std::setw(valsWidths[i]) << std::left << ctxDim0[i]
```

Comment:
Would you need the extra space at the end if you added one to the setw value?

Reply:
:+1: Turns out that I don't!